### PR TITLE
Add proxy config and runCurrentProcessAsUser

### PR DESCRIPTION
**Pull request by hu13**

### CODE REVIEW

File: pvHelpers/proxy_config/pac_parser.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        else:
+            return parse_pac_value(proxy_str, self._proxy_auth)
+
+    def getProxies(self, url, host=None):
+        """
+        :return: a list of proxy dict that can be passed to
+        requests' proxies parameter. Each proxy dict is meant to be used in a
+        fallover fashion. 'None' means the url requires no proxy.
+        """
+
```
Comment (samantehrani) [inline pvHelpers/proxy_config/pac_parser.py]: I think lines 13-20, should happen in `getProxies` if not fetched. `if (pac file not fetched): attempt fetch it`

File: tests/proxy_config/bad_pac_file.pac [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// corrupted pac file that misses FindProxyForURL()
+// function FindProxyForURL(url, host) {
```
[the remark this answers is not in the document]
Comment (hu13) [reply tests/proxy_config/bad_pac_file.pac]: this is intended to mimic bad pac file.

File: pvHelpers/proxy_config/proxy_config_util.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        try:
+            proxy_conf_str = subprocess.check_output(
+                "scutil --proxy", shell=True)
+        except subprocess.CalledProcessError as e:
+            g_log.exception(e)
+            proxy_conf_str = None
+
+    elif "win32" == sys.platform:
+        reg_internet_setting = "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings"
+        ps = "C:\Windows\System32\WindowsPowerShell\\v1.0\powershell.exe"
```
[the remark this answers is not in the document]
Comment (hu13) [reply pvHelpers/proxy_config/proxy_config_util.py]: for window pathname?

File: requirements.txt [hunk 4]
```diff
@@ -6,10 +6,11 @@ https://github.com/PreVeil/vendor/raw/master/python/idna-2.7-py2.py3-none-any.wh
...
 
 https://github.com/PreVeil/vendor/raw/master/python/SQLAlchemy-1.1.5.dev0.tar.gz
-https://github.com/PreVeil/vendor/raw/master/python/Werkzeug-0.11.4-py2.py3-none-any.whl
+https://github.com/PreVeil/vendor/raw/master/python/Werkzeug-0.14.1-py2.py3-none-any.whl
 https://github.com/PreVeil/vendor/raw/master/python/libnacl-1.4.5+preveil1-py2-none-any.whl
 
 https://github.com/PreVeil/vendor/raw/master/python/flanker-0.4.40+preveil1.10.tar.gz
 https://github.com/PreVeil/vendor/raw/master/python/chardet-3.0.4-py2.py3-none-any.whl
 
 https://github.com/PreVeil/vendor/raw/master/python/semver-2.7.7.tar.gz
```
Comment (samantehrani) [inline requirements.txt]: I fixed these recently in master, rebase pls

**Pull request merged**